core-web/apps/dotcms-ui/src/app/portlets/dot-edit-page/seo/components/dot-device-selector-seo/dot-device-selector-seo.component.ts

```diff
@@ -0,0 +1,168 @@
+
+    options: DotDevice[] = [];
+    placeholder = '';
+    socialMediaTiles = [
+        { label: 'Facebook', icon: 'pi pi-facebook' },
+        { label: 'Twitter', icon: 'pi pi-twitter' },
+        { label: 'LinkedIn', icon: 'pi pi-linkedin' }
+    ];
+    defaultOptions: DotDevice[] = [
+        {
+            name: 'Mobile Portrait',
+            icon: 'pi pi-mobile',
+            cssHeight: '390',
+            cssWidth: '844',
+            inode: '0',
+            identifier: ''
+        },
+        {
+            name: 'Mobile Landscape',
+            icon: 'pi pi-mobile',
+            cssHeight: '844',
+            cssWidth: '390',
+            inode: '0',
+            identifier: ''
+        },
+        {
+            name: 'HD Monitor',
+            icon: 'pi pi-desktop',
+            cssHeight: '1920',
+            cssWidth: '1080',
+            inode: '0',
+            identifier: ''
+        },
+        {
+            name: '4K Monitor',
+            icon: 'pi pi-desktop',
+            cssHeight: '3840',
+            cssWidth: '2160',
+            inode: '0',
+            identifier: ''
+        },
+        {
+            name: 'Table Portrait',
+            icon: 'pi pi-tablet',
+            cssHeight: '820',
+            cssWidth: '1180',
+            inode: '0',
+            identifier: ''
+        },
+        {
+            name: 'Table Landscape',
+            icon: 'pi pi-tablet',
+            cssHeight: '1180',
+            cssWidth: '820',
+            inode: '0',
+            identifier: ''
+        }
+    ];
+
+    constructor(
+        private dotDevicesService: DotDevicesService,
+        private dotMessageService: DotMessageService,
+        private readonly cd: ChangeDetectorRef
+    ) {}
+
+    ngOnInit() {
+        this.loadOptions();
+    }
+
+    /**
+     * Track changes in the select
+     * @param DotDevice device
+     */
+    change(device: DotDevice) {
+        this.selected.emit(device);
+    }
+
+    private loadOptions(): void {
+        this.dotDevicesService
+            .get()
+            .pipe(
+                take(1),
+                mergeMap((devices: DotDevice[]) => devices),
+                filter((device: DotDevice) => +device.cssHeight > 0 && +device.cssWidth > 0),
+                toArray(),
+                map((devices: DotDevice[]) =>
+                    this.setOptions(
+                        this.dotMessageService.get('editpage.viewas.default.device'),
+                        devices
+                    )
+                )
+            )
+            .subscribe(
+                (devices: DotDevice[]) => {
+                    this.options = devices;
+                },
+                () => {
+                    this.placeholder = 'No devices';
+                },
+                () => {
+                    this.cd.detectChanges();
+                }
+            );
+    }
+
+    private setOptions(message: string, devices: DotDevice[]): DotDevice[] {
```

Comment:
```suggestion
    private preppendSomething(message: string, devices: DotDevice[]): DotDevice[] {
```

Not sure why this needs to be added tho.